provisioning/service/src/Enrollments/AttestationMechanism.cs

```diff
@@ -56,7 +56,7 @@ private AttestationMechanism(AttestationMechanismType type, TpmAttestation tpm,
                 case AttestationMechanismType.Tpm:
                     if (tpm == null)
                     {
-                        throw new ProvisioningServiceClientException("Invalid TPM attestation mechanism.");
+                        throw new DeviceProvisioningServiceException("Invalid TPM attestation mechanism.");
```

Comment:
What do we set for isTransient? Other properties?

Reply:
This was simply converted from the old `ProvisioningServiceClientException`, so I am not passing other properties here. 

With another glance here, I think we can pass `HttpStatusCode.BadRequest` (400) here and then `IsTransient` will be set as `false` inside the `DeviceProvisioningServiceException` instance. Thoughts?

Reply:
Yeah, that makes sense. I think a unique error code for this case would be even better (in addition to those 2 things).

Reply:
Let's see what enum of common DPS error codes the service team can share with us :)